### ISSUE TITLE
Action causes '[DEP0005] DeprecationWarning'

### ISSUE THREAD

**Issue by dmurat** [gradle/gradle-build-action#22]
For all steps using `gradle-command-action` I'm getting deprecation warning: `[DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.`

Is this expected or should it be fixed?

Tnx

**Comment by dmurat** [comment]
If needed, here is a sample workflow: https://github.com/croz-ltd/klokwrk-project/actions/runs/140909371

**Comment by eskatos** [comment]
Thanks for the report @dmurat.

This deprecation warning is coming from a dependency of the action. I still have to figure out which one.
It isn't a problem per se but that warning will go away when upgrading the offending dependency to a fixed version once available.
I just tried upgrading all outdated dependencies and the deprecation message is still present, I'll get back to this regularly.

**Comment by marracuene** [comment]
Also getting this warning - and it seems to have the ability to intermittently crash the gradle process.

If you look at runs #9 and #10 of [this action](https://github.com/marracuene/playframework/actions?query=workflow%3A%22Run+Gradle+on+PRs%22), you will see that #9 gave this warning and then appeared to fail because of it.    See line `2020-06-23T19:26:06.3961561Z ##[error]Premature close`

Whereas #10 (which differs only in[ the indenting of a comment line ](https://github.com/marracuene/playframework/commit/c82fb90c7efadc3763825dc346177a74480bef8c)- which I used to trigger a new run) did not give this warning, and proceeded with the build.

N.B. this repo is a clone of https://github.com/gradle/playframework, which I forked purely in order to investigate mysterious build failures which I cannot reproduce locally.   The failures only occur in a CI environment.  See: https://github.com/gradle/playframework/issues/136

**Comment by bshaffer** [comment]
Our builds are also failing due to this error, and it does not seem to be intermittent.

**Comment by eskatos** [comment]
Hum, thanks all for the reports! I'll investigate.
In the meantime you can pin your action usage to `v1.2.0` that didn't have this problem.

**Comment by marracuene** [comment]
> In the meantime you can pin your action usage to `v1.2.0` that didn't have this problem.

Hmm.   Pinning to v1.2.0 didn't actually fix it.   Which suggests that something more fundamental is at play.
See [here](https://github.com/marracuene/playframework/runs/811475620?check_suite_focus=true)
... specifically the bit where it says: 
```
Run eskatos/gradle-command-action@v1.2.0
(node:2754) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Cache Size: ~193 MB (202061936 B)
```

**Comment by eskatos** [comment]
I tracked this down to `microsoft/typed-rest-client`.
This has been fixed upstream https://github.com/microsoft/typed-rest-client/commit/de7bcee0527df84f9feadd39d07ad1eba9cff93f but not released yet.

**Comment by nuhkoca** [comment]
Any update on this?

**Comment by amotl** [comment]
Hi again,

@damccorm was so kind to publish `typed-rest-client 1.8.1` [1] right away. Thank you!

With kind regards,
Andreas.

[1] https://www.npmjs.com/package/typed-rest-client

**Comment by bigdaz** [comment]
This was fixed in `v1.4.0`
